### PR TITLE
Add Text Module

### LINKED ISSUE #16: Color Background Text Module

### PR DESCRIPTION
**Pull request by chrismekim**
### Description

**NOTE: We are not adding the bottom text indented to the right option yet, seen in the `mulberry` variant. This was decided to be a nice to have option that I can work on last, when the site is finished. Also, don't pay too much attention to padding here because it will likely change once I confirm some things with XXIX!** 

- Adds Text Module with 2 variants: `default` and `mulberry`
- Also closes #16 
- Updates Founders Impact Slideshow module (adds subheading to schema) 

### Type(s) of changes

- [x] New feature

### Motivation for PR

#18 
#16 
### How Has This Been Tested?

View at `/` to see the `default` variants and `mulberry` variant

### Applicable screenshots:

Default variant
image

Mulberry variant
image

### Follow-up PR

- Added a to-do to work on the button hover styling in a future PR

### CODE REVIEW

**Comment by limsohee1002** [comment]
Need RB

**Comment by chrismekim** [comment]
@limsohee1002 @mokaymokay rebased :)

File: src/components/layouts/TextModule.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        `TextModule TextModule--style-${variant} ${className} bg-color-${variant} ${fontColor}`
+      )}
+    >
+      <div
+        className={cx(
+          `TextModule--style-${variant}--inner-container TextModule__inner-container mxauto col-12 lg:col-8 xxl:col-7 px_75 lg:px1_5`
+        )}
+      >
+        {subheading ? (
+          <span className={cx(`TextModule__subheading primary-sm pr3_75 vertical-align-middle`)}>
```
Comment (limsohee1002) [inline src/components/layouts/TextModule.tsx:40]: It doesn't seem to need `cx` and string interpolation

**Review by limsohee1002 — APPROVED** [review]

File: src/styles/components/TextModule.scss [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  &__button {
+    @extend .transition-shorter;
+  }
+
+  &__button-line-icon {
+    @extend .transition-shorter;
+    width: 1.0625rem;
+    height: 0.125rem;
+
+    &--style-hover {
```
Comment (mokaymokay) [inline src/styles/components/TextModule.scss:86]: I think we can get rid of state in the text module if we do something like:
```
&__button {
  &:hover {
     &-line-icon {
       width: 2.25rem;
     }
  }
}

```

**Review by mokaymokay — CHANGES_REQUESTED** [review]

File: src/styles/components/TextModule.scss (hunk 2 above)
Comment (chrismekim) [reply src/styles/components/TextModule.scss:86]: Like we talked about, I'm adding a to-do to change this after my current PRs are merged so I can change it all at once 😃

**Pull request merged**